### PR TITLE
Added functionality Add All and Clear All

### PR DESCRIPTION
**Pull request by ppollono**
Add Select All and Clear All functionality for multiple selection

### CODE REVIEW

**Comment by israelpereira** [comment]
I need this functionality too,

when it will be merged to the master branch ?

**Comment by israelpereira** [comment]
@brianfeister

**Comment by ppollono** [comment]
@brianfeister, let me know if I need to update something in my PR

**Comment by maikdiepenbroek** [comment]
That's my changes, you're using. I've submitted the exact same PR

**Comment by ppollono** [comment]
I fixed it because I need it for a site, It's almost your code. compare it

**Comment by maikdiepenbroek** [comment]
It's fine by me :+1: Just want it to be merged !

**Comment by ergomesh** [comment]
how can we resolve the conflict? I also need this functionality?

**Comment by ashwin027** [comment]
Is this PR merged? Waiting for this functionality.

**Comment by wesleycho** [comment]
This needs rebasing, as well as tests.

**Comment by ppollono** [comment]
@wesleycho Rebased, will add tests when I get some time for it, using the branch in my project, works fine 😸

File: dist/select.js [hunk 1]
```diff
@@ -1595,6 +1600,14 @@ uis.directive('uiSelectMultiple', ['uiSelectMinErr','$timeout', function(uiSelec
         $selectMultiple.updateModel();
       });
 
+      scope.$on('uis:selectBatch', function (event, items) {
+        if($select.selected.length >= $select.limit) {
+          return;
+        }
+        Array.prototype.push.apply($select.selected, items);
+        $selectMultiple.updateModel();
+      });
```
Comment (ppollono) [inline dist/select.js]: 🏆

**Comment by ppollono** [comment]
rebased, but continue this feature here https://github.com/angular-ui/ui-select/pull/1751, we use the master branch in my app

**Pull request closed without merging**